src/core/Akka.Cluster/Cluster.cs

```diff
@@ -563,22 +564,38 @@ internal void Shutdown()
 
         private readonly IActorRef _clusterDaemons;
         private IActorRef _clusterCore;
+        private Task<IActorRef> _clusterCoreTask;
 
         /// <summary>
         /// TBD
         /// </summary>
+        [Obsolete("use TellCoreSafe()")]
         internal IActorRef ClusterCore
         {
             get
             {
-                if (_clusterCore == null)
+                if (_clusterCore is null)
```

Comment:
Just out of curiosity, is this semantically different than what we did before?

Reply:
`==` can use the operator overload and is most of the time inlining the operator code
`is null` and the new ´is not null` get converted to comparison  simple `Object.ReferenceEquals(obj, null)`
For interface types it can be even worse.